File: swift-solutions/Sources/Slns0kto1k/S0001TwoSum.swift
public func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    var map = [Int: Int]()
    for (i, e) in nums.enumerated() {
        if let v = map[target - e] {
            return [v, i]
        }
        map[e] = i
    }
    return []
}

File: cpp-solutions/src/slns0kto1k/s0001_twosum.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

#include "slns0kto1k.h"


/**
 * given a array with a bunch of numbers and a target numbers
 * find the indexs of the two numbers which num
 */

using namespace std;

vector<int> twoSum(const vector<int> &nums, const int target)
{
  unordered_map<int, int> map;

  for (int i = 0; i < nums.size(); i++) {
    auto iterator = map.find(target - nums[i]);
    if (iterator != map.end()) { return { iterator->second, i }; }
    map[nums[i]] = i;
  }

  return {};
}

void printVector(const vector<int> &vec)
{
  for (auto &&idx : vec) { cout << idx << ' '; }
  cout << '\n';
}
